crates/runtime/src/instance.rs

```diff
@@ -615,19 +488,17 @@ impl Instance {
     ///
     /// Returns `None` if memory can't be grown by the specified amount
     /// of pages.
-    pub(crate) fn memory_grow(
-        &mut self,
-        memory_index: DefinedMemoryIndex,
-        delta: u32,
-    ) -> Option<u32> {
+    pub(crate) fn memory_grow(&self, memory_index: DefinedMemoryIndex, delta: u32) -> Option<u32> {
         let result = self
             .memories
-            .get_mut(memory_index)
+            .get(memory_index)
             .unwrap_or_else(|| panic!("no memory for index {}", memory_index.index()))
             .grow(delta);
 
         // Keep current the VMContext pointers used by compiled wasm code.
-        *self.memory_mut(memory_index) = self.memories[memory_index].vmmemory();
+        unsafe {
+            *self.memory_ptr(memory_index) = self.memories[memory_index].vmmemory();
+        }
```

Comment:
Instead of handing out a `*mut VMMemoryDefinition` and doing an `unsafe` store here in relatively high-level code, what if we had a `set_memory` method which takes care of both the unsafe pointer arithmetic and the unsafe store all in one place?

Reply:
Sounds good to me!